### PR TITLE
Dirty multi retention store

### PR DESCRIPTION
**Pull request by alejandroperez**
Quick and dirty implementation of a multi- retention/data store, that uses individual retentions.

### CODE REVIEW

**Comment by jingwei** [comment]
Spike, what is the reason to change RetentionStoreReader from an interface to a class? In my view, the interface is more flexible. There could be different implementations (like SimpleRetentionStoreReader or CompositeRetentionStoreReader). It is better to keep it as an interface and add an abstract class such as AbstractRetentionStoreReader for code reuse.

Thanks.

**Comment by alejandroperez** [comment]
New commit containing the mod. suggested by jingwei.

File: krati-main/src/retention/java/krati/retention/CompositePosition.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    }
+    
+    @Override
+    public int getIndex() {
+        return positions[0].getIndex();
+    }
+    
+    @Override
+    public boolean isIndexed() {
+        //O solo el primero.
```
Comment (tingchen) [inline krati-main/src/retention/java/krati/retention/CompositePosition.java:74]: is this in Spanish? need English translation

File: krati-main/src/retention/java/krati/retention/CompositePosition.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    public long getOffset() {
+        return positions[0].getOffset();
+    }
+    
+    @Override
+    public int getIndex() {
+        return positions[0].getIndex();
+    }
+    
+    @Override
```
Comment (tingchen) [inline krati-main/src/retention/java/krati/retention/CompositePosition.java:72]: Need some comments here. Overall, I find a single "index" value  for a multiple retention store not natural: that's why we have hard coded 0 in line 69 and 64.  Ideally I want to see different index values for diff retention stores -- however, that will change the interface.

**Comment by jingwei** [comment]
pull request merged.

**Pull request closed without merging**